yazi-plugin/src/utils/user.rs

```diff
@@ -17,19 +17,31 @@ impl Utils {
 
 		ya.raw_set(
 			"user_name",
-			lua.create_function(|lua, uid: Option<u32>| {
+			lua.create_function(|lua, uid: Option<Option<u32>>| {
+				let user_id = match uid {
+					None => USERS_CACHE.get_current_uid(),
+					Some(None) => return Ok(None),
+					Some(Some(u)) => u,
+				};
+
 				USERS_CACHE
-					.get_user_by_uid(uid.unwrap_or_else(|| USERS_CACHE.get_current_uid()))
+					.get_user_by_uid(user_id)
 					.map(|s| lua.create_string(s.name().as_encoded_bytes()))
 					.transpose()
 			})?,
 		)?;
 
 		ya.raw_set(
 			"group_name",
-			lua.create_function(|lua, gid: Option<u32>| {
+			lua.create_function(|lua, gid: Option<Option<u32>>| {
```

Comment:
In what case would the value passed from Lua be `Option<Option<u32>>`? If you call it with `ya.group_name(nil)`, it would only be an `Option<u32>`.

Reply:
See the match clause right after this line. I did it like this to not break the Lua API: 'group_name()' still returns the name of the current user. However, 'group_name(nil)' now returns 'nil'.

'Option<Option<>>' is how mlua differentiates 'nil' with no argument.

Reply:
This looks really confusing because in Lua, passing 0 arguments and passing one argument as `nil` both result in the parameter value being `nil`:

```lua
function foo(bar)
  print(bar)
end

foo()
foo(nil)
```

Ouptup:

```
nil
nil
```

Reply:
Actually this change is not needed anymore since the current version of `ya.owner(cha)` checks if `uid` is `nil`.

There is a caveat that `ya.user_name(cha.uid)` might now return `"root"` (or the user for `uid 0`) if `cha.uid` is `nil`. The file might not be actually owned by `root`. With that `Option<Option<>>` code the return value would be `nil` in this case.

However, the Lua code I have for `folder.lua` / `ya.lua` is doing the necessary checks currently, so I would be happy to merge my current commit with this `Option<Option<>>` change removed. The caveat only applies to potential user plugins.

Incidentally, Lua is able to differentiate a `nil` and no arguments: https://stackoverflow.com/a/12625444

Reply:
Thanks for the change!

> There is a caveat that ya.user_name(cha.uid) might now return "root" (or the user for uid 0) if cha.uid is nil. 

If `cha.uid` is `nil`, the expected behavior is for `ya.user_name(cha.uid)` to return the current user (not the user with `uid 0`), did you find any inconsistency with the current implementation? If so we need to fix it. 

> Lua is able to differentiate a nil and no arguments

Yeah you're right, I just want to avoid this as much as possible because it requires the caller to explicitly distinguish between `ya.user_name()` and `ya.user_name(nil)`, I find it often confusing.

Reply:
Yeah sorry, it returns the current user instead of `root`.